packages/core/src/browser/tree/tree.ts

```diff
@@ -65,14 +65,20 @@ export interface TreeNode {
     readonly id: string;
     /**
      * A human-readable name of this tree node.
+     *
+     * @deprecated use `LabelProvider.getName` instead or move this property to your tree node type
```

Comment:
- [ ] document how `LabelProviderContribution` should be used to customize appearance of the tree node across the application